iModelCore/Units/src/Formatting/NumericFormatSpec.cpp

```diff
@@ -959,4 +963,46 @@ double NumericFormatSpec::RoundDouble(double dval, double roundTo)
     return (dval < 0.0) ? -rnd : rnd;
     }
 
+Utf8String NumericFormatSpec::FormatToFractionalRatio(double dval, Utf8String unitName) const
+{
+    double reciprocal = 1.0 / dval;
+
+    Utf8String result;
+    if (unitName == "VERTICAL_PER_HORIZONTAL")
+        result = "1:" + Format(reciprocal);
+    else if (unitName == "HORIZONTAL_PER_VERTICAL")
+        result = Format(reciprocal) + ":1";
+        
+    return result;
+}
+
+Utf8String NumericFormatSpec::FormatToIntegerRatio(double dval, Utf8String unitName) const
+{
+    double reciprocal = reciprocal = 1.0 / dval;
+
+    double precisionFactor = GetDecimalPrecisionFactor();
+
+    int numerator = static_cast<int>(reciprocal * precisionFactor);
+    int denominator = static_cast<int>(precisionFactor);
+
+    // int gcd = std::gcd(numerator, denominator);^M
+    int a = numerator;
+    int b = denominator;
+    while (b != 0) {
+        int temp = b;
+        b = a % b;
+        a = temp;
+    }
+    
+    numerator /= a;
+    denominator /= a;
+
+    if (unitName == "VERTICAL_PER_HORIZONTAL")

```

Comment:
same as before no hard coded units.
Maybe it helps to imagine it with different units than slope. You could have the input 25 for the "kilometers_per_hour" unit. I think the output should then be "25:1" and if you have an inverted unit for that, the magnitude would be converted into 0.04 (1 / 25), so the generated result should be "1:25"
I believe you can make this depend on whether the dval is bigger or smaller than 1.

Reply:
But let's sync up with Simon about this. He can provide more and better examples. I'm not the expert for ratios...